sbtc-core/src/operations/commit_reveal/construction.rs

```diff
@@ -0,0 +1,157 @@
+/*!
+Construction of commit reveal transactions
+*/
+use std::iter::{once, repeat};
+
+use bdk::bitcoin::{
+    secp256k1::ecdsa::RecoverableSignature, Address as BitcoinAddress, Transaction, TxOut,
+    XOnlyPublicKey,
+};
+use stacks_core::address::StacksAddress;
+
+use crate::operations::{
+    commit_reveal::utils::{commit, reveal, CommitRevealError, CommitRevealResult, RevealInputs},
+    Opcode,
+};
+
+/// Data to construct a commit reveal deposit transaction
+pub struct DepositData<'p> {
+    /// Address to deposit to
+    pub address: &'p StacksAddress,
+    /// Name of the contract to deposit to
+    pub contract_name: Option<&'p str>,
+    /// How much to send for the reveal fee
+    pub reveal_fee: u64,
+}
+
+impl<'p> DepositData<'p> {
+    /// Create deposit data
+    pub fn new(
+        address: &'p StacksAddress,
+        contract_name: Option<&'p str>,
+        reveal_fee: u64,
+    ) -> Self {
+        Self {
+            address,
+            contract_name,
+            reveal_fee,
+        }
+    }
+
+    /// Serializes this data according to the SIP-021 wire formats
+    /// Links:
+    ///  - [SIP draft](https://github.com/stacksgov/sips/blob/56b73eada5ef1b72376f4a230949297b3edcc562/sips/sip-021/sip-021-trustless-two-way-peg-for-bitcoin.md)
+    ///  - [Reference implementation](https://github.com/stacks-network/stacks-blockchain/blob/next/src/chainstate/burn/operations/peg_in.rs)
```

Comment:
SIP-021 is deprecated. I think it's better to point to the sBTC docs. This is the new reference implementation ;)

Reply:
Will do 👍